graphql/schemabuilder/pagination.go

```diff
@@ -393,55 +395,96 @@ func (c *connectionContext) pagesFromEdges(edges []Edge, limit int) (pages []str
 	return pages
 }
 
-func (c *connectionContext) applyTextFilter(ctx context.Context, nodes []interface{}, args PaginationArgs) ([]interface{}, error) {
-	if args.FilterText == nil || *args.FilterText == "" {
-		return nodes, nil
-	}
+func (c *connectionContext) applyBatchTextFilter(ctx context.Context, nodes []interface{}, args PaginationArgs) ([]bool, error) {
 
 	nodesToKeep := make([]bool, len(nodes))
 
 	g, ctx := errgroup.WithContext(ctx)
-	for unscopedI, unscopedNode := range nodes {
-		i, node := unscopedI, unscopedNode
-		g.Go(func() error {
-			keep := false
-			for name, filterField := range c.FilterTextFields {
-				// Resolve the graphql.Field made for sorting.
-				text, err := filterField.Resolve(ctx, node, nil, nil)
+	g.Go(func() error {
+		for name, filterField := range c.FilterTextFields {
+			// Resolve the graphql.Field made for sorting.
```

Comment:
maybe we could do it once for each filterField though.

Reply:
Do a goroutine to get all the FilterField results in parallel, wait for them to finish, and filter all the text at the same time.